cro-clib/src/types.rs

```diff
@@ -46,7 +64,43 @@ impl CroResult {
     }
 }
 
+/// copy from c array to rust array
+/// # Safety
+pub unsafe fn copy_string(src: &str, dst: &mut [u8]) {
+    dst[..src.len()].copy_from_slice(&src.as_bytes()[..src.len()]);
+    dst[src.len()] = 0;
+}
+
 /// # Safety
 pub unsafe fn get_string(src: *const c_char) -> String {
     CStr::from_ptr(src).to_string_lossy().into_owned()
 }
+
+/// value: carson unit  for example) 1_0000_0000 carson = 1 cro, 1 carson = 0.0000_0001 cro
+#[repr(C)]
+pub struct CroUtxo {
+    pub address: [u8; 32], // holder
+    pub value: u64,        // amount in carson unit
+    pub valid_from: u64,   // time lock
+}
+
+#[derive(Clone, Default)]
+pub struct CroTx {
+    pub txin: Vec<TxOut>,
```

Comment:
what is the purpose of `txin` ? It adds extra information, but I don't see it used anywhere -- only the length is used? Having only `tx.inputs` should be enough?

Reply:
this is TxoPointer's utxo, supplied from user.

Reply:
but why not to hold in in `CroTx.tx` ?